source/d3d9/runtime_d3d9.cpp

```diff
@@ -1236,19 +1471,149 @@ namespace reshade::d3d9
 
 	void runtime_d3d9::draw_debug_menu()
 	{
+		ImGui::Text("MSAA is %s", _is_multisampling_enabled ? "active" : "inactive");
+		ImGui::Spacing();
+
 		if (ImGui::CollapsingHeader("Buffer Detection", ImGuiTreeNodeFlags_DefaultOpen))
 		{
-			if (ImGui::Checkbox("Disable replacement with INTZ format", &_disable_intz))
+			if (ImGui::Checkbox("Preserve depth buffer from being cleared", &_preserve_depth_buffer))
+			{
+				runtime::save_config();
+
+				// Force depth-stencil replacement recreation
+				_depthstencil = nullptr;
+				// Force depth-stencil clearing table recreation
+				_depth_buffer_table.clear();
+				_depth_clearing_table.clear();
+			}
+
+			ImGui::Spacing();
+			ImGui::Spacing();
+
+			if (!_preserve_depth_buffer)
+			{
+				if (ImGui::Checkbox("Disable replacement with INTZ format", &_disable_intz))
+				{
+					runtime::save_config();
+
+					// Force depth-stencil replacement recreation
+					_depthstencil = nullptr;
+				}
+			}
+
+			ImGui::Spacing();
+
+			if (ImGui::Checkbox("Disable the depth buffer size restriction", &_disable_depth_buffer_size_restriction))
 			{
 				runtime::save_config();
 
 				// Force depth-stencil replacement recreation
 				_depthstencil = nullptr;
+				// Force depth-stencil clearing table recreation
+				_depth_buffer_table.clear();
+				_depth_clearing_table.clear();
 			}
 
-			for (const auto &it : _depth_source_table)
+			if (_preserve_depth_buffer)
+			{
+				if (ImGui::Checkbox("multiple depthstencil replacement buffer", &_multi_depthstencil))
```

Comment:
Does this help in some games and why? It adds a lot of extra complication.

Reply:
In some games, it helped (for instance Oblivion), but we can wait to implement it